inst/Examples/2ArmTimeToEventOutcomeAnalysis/RCode/AnalyzeUsingHazardRatioLimitsOfCI.R

```diff
@@ -0,0 +1,138 @@
+AnalyzeUsingHazardRatioLimitsOfCI <- function(SimData, DesignParam, LookInfo = NULL, UserParam = NULL)
+{   
+    
+    # Input objects can be saved through the following lines:
+    
+    # setwd( ["Set the path to save the .Rds files"])
+    # saveRDS( SimData, "SimData.Rds")
+    # saveRDS( DesignParam, "DesignParam.Rds" )
+    # saveRDS( LookInfo, "LookInfo.Rds" )
+
+   
+    if( !is.null( LookInfo ) )
+    {
+        # Look info was provided so use it
+        nLookIndex           <- LookInfo$CurrLookIndex
+        nQtyOfEvents         <- LookInfo$CumEvents[ nLookIndex ]
+        dEffBdry             <- LookInfo$EffBdryLower[ nLookIndex ]
+    }else
+    {   # Look info is not provided for fixed sample designs so fetch the information appropriately
+        nLookIndex           <- 1
+        nQtyOfEvents         <- DesignParam$MaxEvents
+        dEffBdry             <- DesignParam$CriticalPoint
+    }
+    
+    SimData$TimeOfEvent      <- SimData$ArrivalTime + SimData$SurvivalTime    # This is the calendar time in the trial that the patients event is observed
+    
+    # Compute the time of analysis 
+    SimData                  <- SimData[ order( SimData$TimeOfEvent), ]
+    dTimeOfAnalysis          <- SimData[ nQtyOfEvents, ]$TimeOfEvent
+    
+    # Add the Observed Time variable 
+    SimData                  <- SimData[ SimData$ArrivalTime <= dTimeOfAnalysis ,]   # Exclude any patients that were not enrolled by the time of the analysis
+    SimData$Event            <- ifelse( SimData$TimeOfEvent > dTimeOfAnalysis, 0, 1 )  # If the event is observed after the analysis it is not observed, eg censored 
+    SimData$ObservedTime     <- ifelse( SimData$TimeOfEvent > dTimeOfAnalysis, dTimeOfAnalysis - SimData$ArrivalTime, SimData$TimeOfEvent - SimData$ArrivalTime )
+    
+    # Order the data by observed time for the remainder of the computations
+    SimData                  <- SimData[ order( SimData$ObservedTime ), ]
+    
+    # Compute Observed HR
+    coxModel                 <- coxph( Surv( ObservedTime, Event ) ~ TreatmentID, data = SimData )
+    
+    # find the value of observed Log(HR) and SE(log(HR))
+    dlogHR                   <- summary( coxModel )$coefficients[ 1 ]
+    dstdError                <- summary( coxModel )$coefficients[ 3 ]
+    
+    # Log HR follows Normal distribution with mean = observed log HR on line no 83 and Std error given on line no 84
+    # Critical value for Z test is given as,
+    dalpha                   <- 1 - UserParam$dConfLevel
+    dZalpha                  <- qnorm( 1 - dalpha/2 )
+    
+    # Confidence Interval for log HR is given as,
+    
+    dLowerLimitCI                   <- dlogHR - dZalpha * dstdError
+    dUpperLimitCI                   <- dlogHR + dZalpha * dstdError
+    
+    
+    # A decision of 2 means success, 0 means continue the trial
+    nDecision                <- ifelse( dLowerLimitCI < log(UserParam$dMAV), 2, 0 )  # log HR scale

```

Comment:
Please see note about the rule for HR and correct in this example as well.  Please let me know if you have any questions.

Reply:
updated as per the rule mentioned above